pytket/binders/passes.cpp

```diff
@@ -355,9 +355,23 @@ PYBIND11_MODULE(passes, m) {
       "Converts all multi-qubit gates into CX and single-qubit gates.");
   m.def(
       "GlobalisePhasedX", &GlobalisePhasedX,
-      "Replaces every occurence of PhasedX or NPhasedX gates with NPhasedX "
-      "gates acting on all qubits, and correcting rotation gates, so that the "
-      "GlobalPhasedXPredicate is satisfied.");
+      "Turns all PhasedX and NPhasedX gates into global gates\n\n"
+      "Replaces any PhasedX gates with global NPhasedX gates. "
+      "By default, this transform will squash all single-qubit gates "
+      "to PhasedX and Rz gates before proceeding further. "
+      "Existing non-global NPhasedX will not be preserved."
+      "This is the recommended setting for best "
+      "performance. If squashing is disabled, each non-global PhasedX gate "
+      "will be replaced with 2x global NPhasedX, but any other gates will "
+      "be left untouched."
+      "\n\n:param squash: Whether to squash the circuit in pre-processing "
+      "(default: true)."
+      "\n\nIf squash=true (default), this transform always returns true. "
```

Comment:
In a way, this is like `PauliSimp` always returning true, even if it could be that the end circuit is identical to the initial one.

Short of comparing the whole circuit pre- and post-pass, there is currently no way to check for changes: squashing will decompose and merge any NPhasedX gates into TK1, and then resynthesise global gates out of this. It might be that after decompsing global gates and resynthesing them, we land back on our feet at the end.

We could return false in a few simple cases (like when there are no global gates at all), but I think this would just make the semantics of that return value more obscure.

I considered not decomposing gates when they are already global, but in many cases this wouldn't be performant. I suggest that users that want more control over this just use `squash=false`. Otherwise, we could add an additional flag `preserve_global`, but I'm not convinced this is required at the moment.

Reply:
Yes indeed, this is the return value of `apply`. I will address this in a commit